pkg/restic/repository_manager.go

```diff
@@ -205,6 +213,143 @@ func (rm *repositoryManager) Forget(snapshot SnapshotIdentifier) error {
 	return rm.exec(cmd)
 }
 
+func (rm *repositoryManager) ChangeKey(name string) error {
+	log := rm.log.WithField("repo", name)
+	log.Debug("Changing key")
+
+	repo, err := getReadyRepo(rm.repoLister, rm.namespace, name)
+	if err != nil {
+		return err
+	}
+
+	rm.repoLocker.LockExclusive(name)
+	defer rm.repoLocker.UnlockExclusive(name)
+
+	secret, err := rm.secretsLister.Secrets(name).Get(CredentialsSecretName)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// if there's data at the NewCredentialsKey within the secret, it could mean any of the following:
+	//	- this is our first time processing this key-change request
+	//	- we attempted to change the repo key already but got an error
+	//  - we successfully changed the repo key, but failed to update the secret to
+	//	  reflect the change
+	if newKey, ok := secret.Data[NewCredentialsKey]; ok {
+		log.Debugf("Found key in secret at %s", NewCredentialsKey)
+
+		if string(newKey) == string(secret.Data[CurrentCredentialsKey]) {
+			log.Debugf("New key is the same as current key")
+
+			return patchSecret(secret, func(s *corev1api.Secret) {
+				delete(s.Data, NewCredentialsKey)
+			}, rm.secretsClient)
+		}
+
+		newKeyFile, err := writeTempFile("", "", newKey)
+		if err != nil {
+			return err
+		}
+		defer os.Remove(newKeyFile)
+
+		// if it's not valid, we need to run the restic cmd to change repo keys
+		if !isKeyValid(repo.Spec.ResticIdentifier, newKeyFile) {
+			log.Debugf("Key in %s is not valid", NewCredentialsKey)
```

Comment:
Per slack, it's probably worth using specific log messages as to why it's not valid.